runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SchemaTranslation.java

```diff
@@ -25,33 +25,29 @@
 import org.apache.beam.sdk.schemas.Schema.FieldType;
 import org.apache.beam.sdk.schemas.Schema.LogicalType;
 import org.apache.beam.sdk.schemas.Schema.TypeName;
-import org.apache.beam.sdk.util.SerializableUtils;
-import org.apache.beam.vendor.grpc.v1p13p1.com.google.protobuf.ByteString;
 import org.apache.beam.vendor.guava.v20_0.com.google.common.collect.BiMap;
 import org.apache.beam.vendor.guava.v20_0.com.google.common.collect.ImmutableBiMap;
 import org.apache.beam.vendor.guava.v20_0.com.google.common.collect.Maps;
 
 /** Utility methods for translating schemas. */
 public class SchemaTranslation {
-  private static final BiMap<TypeName, RunnerApi.Schema.TypeName> TYPE_NAME_MAPPING =
-      ImmutableBiMap.<TypeName, RunnerApi.Schema.TypeName>builder()
-          .put(TypeName.BYTE, RunnerApi.Schema.TypeName.BYTE)
-          .put(TypeName.INT16, RunnerApi.Schema.TypeName.INT16)
-          .put(TypeName.INT32, RunnerApi.Schema.TypeName.INT32)
-          .put(TypeName.INT64, RunnerApi.Schema.TypeName.INT64)
-          .put(TypeName.DECIMAL, RunnerApi.Schema.TypeName.DECIMAL)
-          .put(TypeName.FLOAT, RunnerApi.Schema.TypeName.FLOAT)
-          .put(TypeName.DOUBLE, RunnerApi.Schema.TypeName.DOUBLE)
-          .put(TypeName.STRING, RunnerApi.Schema.TypeName.STRING)
-          .put(TypeName.DATETIME, RunnerApi.Schema.TypeName.DATETIME)
-          .put(TypeName.BOOLEAN, RunnerApi.Schema.TypeName.BOOLEAN)
-          .put(TypeName.BYTES, RunnerApi.Schema.TypeName.BYTES)
-          .put(TypeName.ARRAY, RunnerApi.Schema.TypeName.ARRAY)
-          .put(TypeName.MAP, RunnerApi.Schema.TypeName.MAP)
-          .put(TypeName.ROW, RunnerApi.Schema.TypeName.ROW)
-          .put(TypeName.LOGICAL_TYPE, RunnerApi.Schema.TypeName.LOGICAL_TYPE)
+
+  private static final BiMap<TypeName, RunnerApi.Schema.AtomicType> ATOMIC_TYPE_MAPPING =
+      ImmutableBiMap.<TypeName, RunnerApi.Schema.AtomicType>builder()
+          .put(TypeName.BYTE, RunnerApi.Schema.AtomicType.BYTE)
+          .put(TypeName.INT16, RunnerApi.Schema.AtomicType.INT16)
+          .put(TypeName.INT32, RunnerApi.Schema.AtomicType.INT32)
+          .put(TypeName.INT64, RunnerApi.Schema.AtomicType.INT64)
+          .put(TypeName.FLOAT, RunnerApi.Schema.AtomicType.FLOAT)
+          .put(TypeName.DOUBLE, RunnerApi.Schema.AtomicType.DOUBLE)
+          .put(TypeName.STRING, RunnerApi.Schema.AtomicType.STRING)
+          .put(TypeName.BOOLEAN, RunnerApi.Schema.AtomicType.BOOLEAN)
+          .put(TypeName.BYTES, RunnerApi.Schema.AtomicType.BYTES)
           .build();
 
+  private static final String URN_BEAM_LOGICAL_DATETIME = "urn:beam:logical:datetime";
```

Comment:
Two stylistic nits:

1. We tend to omit the `urn`, don't we? While it does make a valid URI out of the thing, it seems a bit silly.
2. I would leave out `logical` but put in something like `type` or `schema_type` or `fieldtype` to namespace.

Reply:
Done. I went with fieldtype